scripts/check-docs.sh

```diff
@@ -10,38 +8,39 @@ shopt -s globstar
 #   Checks that the contents of /docs folder are valid. This includes:
 #
 #     1. Ensuring the the .cue files can compile.
-#     2. Link validation.
+#     2. In CI, ensuring the the .cue files are properly formatted.
 
-DOCS_PATH="docs"
+cd "$(dirname "${BASH_SOURCE[0]}")/.."
 
-if ! [ -x "$(command -v cue)" ]; then
-  echo 'Error: cue is not installed.' >&2
+read-all-docs() {
+  scripts/cue.sh list | sort | xargs cat -A
+}
+
+if ! cue version >/dev/null; then
+  echo 'Error: cue is not installed'
   exit 1
 fi
 
-if [[ -z "${CI:-}" ]]; then
-  echo "Skipping local formatting - reserved for CI"
+if [[ "${CI:-"false"}" != "true" ]]; then
```

Comment:
Personally, I would prefer to keep the format check locally. I find that it saves me more time than seeing CI errors later on and coming back and fixing.

Reply:
The problem is that, until there is `cue fmt --check`, we have to perform the formatting and then compare the files before and after the formatting to see if the format is correct.

Running in locally would be ok if it wouldn't alter the files, however it currently reformats files in-place. This is not good, because it is possible that the user runs this check on a work-in-progress tree, and formatting the files disrupts the work. It can even corrupt the files in some cases.

Due to issues with the check-docs, I usually end up invoking `cue` manually, and this PR in part helps with that flow: I'll likely be using `scripts/cue.sh fmt && scripts/cue.sh eval -e 'what.im.working.on'` or sth